storybookjs/storybook
# UI: Display menu icon on the toolbar when the sidebar is collapsed

Issue: #13799

## What I did
I added a new `ToolbarMenu` type in the `Menu.tsx` file and created a new `menu` tool, then added the tool in the left side of the toolbar. I also added a new story in the `Menu.stories.tsx` file and fixed two stories that had some errors. However, there are some problems, which I list below and this is the reason why I created a draft pull request initially. 

* The display of the icon is only successful whenever you load a preview story with the sidebar collapsed. Then, using the shortcut `S` you can make the sidebar appear again and the icon disappears and appears again when you use the shortcut. Navigation from the menu is not successful. To make the menu icon appear, I used the `state.layout.showNav` property, but maybe this is not the right way and this is the cause of the above problems. However, I noticed that the above bugs only occurred in my instance of storybook, whereas the icon is displayed as intended in the deploy preview.
* I also planned to add a new preview story where the icon is visible, however there were some changes in the `preview.tsx` file in the `components` module and I think that now all of the preview stories should be modified.

## How to test

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps?
I added new menu stories, which are visible in the `official-example` app.
- Does this need an update to the documentation?
I am not sure but I can update the documentation if needed.

If your answer is yes to any of these, please make sure to include it in your PR.
